packages/block-library/src/block/edit.js

```diff
@@ -6,207 +6,188 @@ import { partial } from 'lodash';
 /**
  * WordPress dependencies
  */
-import { Component } from '@wordpress/element';
-import { Placeholder, Spinner, Disabled } from '@wordpress/components';
-import { withSelect, withDispatch } from '@wordpress/data';
-import { __ } from '@wordpress/i18n';
 import {
 	BlockEditorProvider,
 	BlockList,
 	WritingFlow,
 } from '@wordpress/block-editor';
-import { compose } from '@wordpress/compose';
 import { parse, serialize } from '@wordpress/blocks';
+import { Placeholder, Spinner, Disabled } from '@wordpress/components';
+import { useDispatch, useSelect } from '@wordpress/data';
+import { useEffect, useState } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
 
 /**
  * Internal dependencies
  */
 import ReusableBlockEditPanel from './edit-panel';
 
-class ReusableBlockEdit extends Component {
-	constructor( { reusableBlock } ) {
-		super( ...arguments );
-
-		this.startEditing = this.startEditing.bind( this );
-		this.stopEditing = this.stopEditing.bind( this );
-		this.setBlocks = this.setBlocks.bind( this );
-		this.setTitle = this.setTitle.bind( this );
-		this.save = this.save.bind( this );
-
-		if ( reusableBlock ) {
-			// Start in edit mode when we're working with a newly created reusable block
-			this.state = {
-				isEditing: reusableBlock.isTemporary,
-				title: reusableBlock.title,
-				blocks: parse( reusableBlock.content ),
-			};
-		} else {
-			// Start in preview mode when we're working with an existing reusable block
-			this.state = {
-				isEditing: false,
-				title: null,
-				blocks: [],
+const EMPTY_ARRAY = [];
+
+export default function ReusableBlockEdit( { attributes, isSelected } ) {
+	const { ref } = attributes;
+
+	const {
+		theReusableBlock,
```

Comment:
I see the problem you're addressing here. Can't say I love the "the" prefix as a generalized solution. I guess it'll continue to be an issue we deal with, save for some other prior attempts to work around this (#21057, [#18973](https://github.com/WordPress/gutenberg/pull/18973/files#diff-2a00f67990de843497e20b203dcf4084R23)). The other option I might resort to is something "less ideal" inside the selector, like an underscore prefix `_`, so at least the desirable name `reusableBlock` is granted for the primary usage scope.

Reply:
The underscore pattern seems pretty reasonable to me; I'll use that.